test/Microsoft.AspNet.Localization.Tests/QueryStringRequestCultureProviderTest.cs

```diff
@@ -0,0 +1,176 @@
+                    var requestCulture = requestCultureFeature.RequestCulture;
+                    Assert.Equal(options.DefaultRequestCulture.Culture.Name, requestCulture.Culture.Name);
+                    Assert.Equal(options.DefaultRequestCulture.UICulture.Name, requestCulture.UICulture.Name);
+                    return Task.FromResult(0);
+                });
+            },
+            services =>
+            {
+                services.AddLocalization();
+            }))
+            {
+                var client = server.CreateClient();
+                var response = await client.GetAsync("/page");
+            }
+        }
+
+        [Fact]
+        public async void GetDefaultCultureInfoIfCultureValuesAreInvalid()
+        {
+            using (var server = TestServer.Create(app =>
+            {
+                var options = new RequestLocalizationOptions();
+                app.UseRequestLocalization(options);
+                app.Run(context =>
+                {
+                    var requestCultureFeature = context.GetFeature<IRequestCultureFeature>();
+                    var requestCulture = requestCultureFeature.RequestCulture;
+                    Assert.Equal(options.DefaultRequestCulture.Culture.Name, requestCulture.Culture.Name);
+                    Assert.Equal(options.DefaultRequestCulture.UICulture.Name, requestCulture.UICulture.Name);
+                    return Task.FromResult(0);
+                });
+            },
+            services =>
+            {
+                services.AddLocalization();
+            }))
+            {
+                var client = server.CreateClient();
+                var response = await client.GetAsync("/page?culture=ar-SA&ui-culture=ar-XY");
+            }
+        }
+
+        //TODO: Uncomment if the PR #63 accepted
+        //[Fact]
+        //public async void GetSameCultureInfoIfCultureKeyIsMissing()
+        //{
+        //    using (var server = TestServer.Create(app =>
+        //    {
+        //        var options = new RequestLocalizationOptions();
+        //        app.UseRequestLocalization(options);
+        //        app.Run(context =>
+        //        {
+        //            var requestCultureFeature = context.GetFeature<IRequestCultureFeature>();
+        //            var requestCulture = requestCultureFeature.RequestCulture;
+        //            Assert.Equal("ar-SA", requestCulture.Culture.Name);
+        //            Assert.Equal("ar-SA", requestCulture.UICulture.Name);
+        //            return Task.FromResult(0);
+        //        });
+        //    },
+        //    services =>
+        //    {
+        //        services.AddLocalization();
+        //    }))
+        //    {
+        //        var client = server.CreateClient();
+        //        var response = await client.GetAsync("/page?ui-culture=ar-SA");
+        //    }
+        //}
+
+        [Fact]
+        public async void GetSameCultureInfoIfUICultureKeyIsMissing()
```

Comment:
It's already there, but was commented wait to accept the PR, while it's accepted i just uncomment the code :smile: